### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by 999bits**
### Description
Hi, This is a decentralized voting application on the Ethereum blockchain. 
The application should allow users to create and participate in polls, vote for their preferred options, and display the results in real-time.
### Checklist

-  I have developed it with solidity and run with hardhat.

-  I have run the tests on my machine and testnet

-  I have rebased my changes to the latest version of the dev branch

### CODE REVIEW

File: .gitignore
Comment (canner1) [inline .gitignore:1]: Add .vscode here so that it is not pushed to git

File: .vscode/extensions.json
Comment (canner1) [inline .vscode/extensions.json:1]: Remove .vscode from git

File: README.md [hunk 1]
```diff
@@ -1 +1,13 @@
-# decentralized_voting_app
+# Sample Hardhat Project
```
Comment (canner1) [inline README.md]: Write readme correctly

File: contracts/Voting.sol [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    /*//////////////////////////////////////////////////////////////
+                        Add options to a poll
+    //////////////////////////////////////////////////////////////*/
+    function addOption(
+        uint _pollId,
+        string memory _text
+    ) public onlyCreator(_pollId) {
+        polls[_pollId].optionCount++;
+        polls[_pollId].options[polls[_pollId].optionCount] = Option(
```
Comment (canner1) [inline contracts/Voting.sol:81]: Optimize gas, optionCount is already used by the index, so you don't need to store in Option strucut

File: contracts/Voting.sol [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        emit CreatePoll(msg.sender, _title, _description);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        Add options to a poll
+    //////////////////////////////////////////////////////////////*/
+    function addOption(
+        uint _pollId,
+        string memory _text
+    ) public onlyCreator(_pollId) {
```
Comment (canner1) [inline contracts/Voting.sol:79]: It's good to create a modifier to validate creator.
But you can also validate poolId, if poolId is bigger than poolCount, you can revert.
This way the code will be safer and optimized

File: contracts/Voting.sol [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        );
+        _;
+    }
+
+    event CreatePoll(
+        address indexed creator,
+        string _title,
+        string _description
+    );
+    event AddOption(address indexed creator, uint _pollId, string _text);
```
Comment (canner1) [inline contracts/Voting.sol]: CreatePoll and AddOption looks like a function name.
Rename them to be looked as events. For example, PollCreated or Poll Added

File: contracts/Voting.sol [hunk 5]
```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract Voting {
```
Comment (canner1) [inline contracts/Voting.sol]: The contract name is not appropriate. Use nouns to name the smart contract

File: contracts/Voting.sol [hunk 6]
```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
```
Comment (canner1) [inline contracts/Voting.sol:2]: It's always good to choose solidity version as a specific one.

File: contracts/Voting.sol [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+contract Voting {
+    struct Poll {
+        uint id;
```
Comment (canner1) [inline contracts/Voting.sol]: Optimize gas, id is used for index, so no need to be stored in the struct

File: hardhat.config.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
+require("@nomicfoundation/hardhat-toolbox");
+// require("@nomicfoundation/hardhat-verify");
+require("@nomiclabs/hardhat-etherscan");
+
+const { mnemonic } = require('./secrets.json');
```
Comment (canner1) [inline hardhat.config.js]: Use dotenv and .env file to get mnemonic

File: hardhat.config.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+      url: "https://bsc-dataseed.binance.org/",
+      chainId: 56,
+      gasPrice: 20000000000,
+      accounts: {mnemonic: mnemonic}
+    }
+  },
+  etherscan: {
+    // Your API key for Etherscan
+    // Obtain one at https://bscscan.com/
+    apiKey: "6KQGJ92VSZPMYUEWN3WBXNYFJHHP72Z9N8"
```
Comment (canner1) [inline hardhat.config.js]: Add api key to .env file and import from it. This is private value.

File: secrets.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+    "mnemonic": "head pigeon debate ten curtain olive middle ride hat little hill bring"
+}
```
Comment (canner1) [inline secrets.json]: Are you kidding? REMOVE THIS!
Do not share any private key or mnemonic to online, Hackers everywhere

**Review by canner1 — CHANGES_REQUESTED** [review]

**Comment by canner1** [comment]
Question: Why are you using mappings instead of arrays? Can you specify the reason here?

File: contracts/Voting.sol [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+    //////////////////////////////////////////////////////////////*/
+    function createPoll(
+        string memory _title,
+        string memory _description
+    ) public {
+        require(
+            bytes(_title).length > 0 && bytes(_description).length > 0,
+            "Title and description are required"
+        );
+        pollCount++;
```
Comment (canner1) [inline contracts/Voting.sol:63]: I think this line should be after creating poll, you should start pool index from zero not 1

File: contracts/Voting.sol [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        Add options to a poll
+    //////////////////////////////////////////////////////////////*/
+    function addOption(
+        uint _pollId,
+        string memory _text
+    ) public onlyCreator(_pollId) {
+        polls[_pollId].optionCount++;
```
Comment (canner1) [inline contracts/Voting.sol:80]: Same index issue with the poll count++

File: .gitignore
Comment (999bits) [reply .gitignore:1]: done

File: contracts/Voting.sol (hunk 7 above)
Comment (999bits) [reply contracts/Voting.sol]: Changed id into optionCount 'cause a poll can have several options.

**Comment by 999bits** [comment]
> Question: Why are you using mappings instead of arrays? Can you specify the reason here?
I often used to use mapping variable before, but I could know that mapping waste more gas than array. 
I am going to use array than mapping and suitable variables at certain environment....

File: contracts/Voting.sol (hunk 11 above)
Comment (999bits) [reply contracts/Voting.sol:63]: Should I always start any index from zero? Maybe or not.. anyway, I 'll consider it.

**Review by canner1 — APPROVED** [review]

**Pull request merged**